integration-tests/src/integration-tests/java/oracle/kubernetes/operator/utils/TestUtils.java

```diff
@@ -0,0 +1,544 @@
+        logger.info(
+            "Service is not created Ite ["
+                + i
+                + "/"
+                + MAX_ITERATIONS_POD
+                + "], sleeping "
+                + WAIT_TIME_POD
+                + " seconds more");
+        try {
+          Thread.sleep(WAIT_TIME_POD * 1000);
+        } catch (InterruptedException ignore) {
+        }
+
+        i++;
+      } else {
+        logger.info("Service " + serviceName + " is Created");
+        break;
+      }
+    }
+  }
+  /**
+   * @param propsFile - input props file
+   * @param inputFileTemplate - operator/domain inputs template file
+   * @param inputYamlFilePath - output file with replaced values
+   * @throws Exception
+   */
+  public static void createInputFile(
+      Properties props, String inputFileTemplate, Path inputYamlFilePath) throws Exception {
+    logger.info("Creating input yaml file at " + inputYamlFilePath);
+
+    //copy create-operator-inputs.yaml and modify it
+    Files.copy(
+        new File(inputFileTemplate).toPath(),
+        inputYamlFilePath,
+        StandardCopyOption.REPLACE_EXISTING);
+    inputYamlFilePath.toFile().setWritable(true);
+
+    //read each line in input template file and replace with op props
+    BufferedReader reader = new BufferedReader(new FileReader(inputYamlFilePath.toString()));
+    String line = "";
+    StringBuffer changedLines = new StringBuffer();
+    boolean isLineChanged = false;
+    while ((line = reader.readLine()) != null) {
+      Enumeration enuKeys = props.keys();
+      while (enuKeys.hasMoreElements()) {
+        String key = (String) enuKeys.nextElement();
+        //if a line starts with the props key then replace
+        //the line with key:value in the file
+        if (line.startsWith(key) || line.startsWith("#" + key)) {
+          changedLines.append(key).append(":").append(props.getProperty(key)).append("\n");
+          isLineChanged = true;
+          break;
+        }
+      }
+      if (!isLineChanged) {
+        changedLines.append(line).append("\n");
+      }
+      isLineChanged = false;
+    }
+    reader.close();
+    //writing to the file
+    FileWriter writer = new FileWriter(inputYamlFilePath.toString());
+    writer.write(changedLines.toString());
+    writer.close();
+  }
+
+  public static String getHostName() {
+    return executeCommand(new String[] {"/bin/sh", "-c", "hostname | awk -F. '{print $1}'"}).trim();
+  }
+
+  public static int getClusterReplicas(String domainUid, String clusterName, String domainNS) {
+    StringBuffer cmd = new StringBuffer();
+    cmd.append("kubectl get domain ")
+        .append(domainUid)
+        .append(" -n ")
+        .append(domainNS)
+        .append(" -o jsonpath='{.spec.clusterStartup[?(@.clusterName == \"")
+        .append(clusterName)
+        .append("\")].replicas }'");
+    logger.fine("getClusterReplicas cmd =" + cmd);
+    String output = TestUtils.executeCommand(new String[] {"/bin/sh", "-c", cmd.toString()});
```

Comment:
I think I've seen the /bin/sh -c stuff a bunch of places - how about adding another variant of executeCommand that does this?